integrations/azure-devops/azure_devops/client/azure_devops_client.py

```diff
@@ -507,3 +519,171 @@ async def get_file_by_commit(
+                "POST",
+                items_batch_url,
+                params={"api-version": "7.1"},
+                data=json.dumps(request_data),
+                headers={"Content-Type": "application/json"},
+            )
+
+            if response is None:
+                logger.warning(
+                    f"No response from itemsbatch API for repository {repository['name']}"
+                )
+                return
+
+            if response.status_code == 400:
+                logger.warning(
+                    f"Bad request (400) for repository {repository['name']}: {response.json().get('message')}"
+                )
+                return
+
+            batch_results = response.json()
+
+            # Flatten nested arrays to get a single list of file dictionaries
+            files = [
+                file_info
+                for sublist in batch_results.get("value", [])
+                for file_info in sublist
+            ]
+            logger.info(f"Found {len(files)} files in repository {repository['name']}")
+
+            downloaded_files = await process_in_queue(
+                files,
+                self.download_single_file,
+                repository,
+                branch,
+                concurrency=MAX_CONCURRENT_FILE_DOWNLOADS,
+            )
+
+            for file in downloaded_files:
+                yield [file]
+
+        except HTTPStatusError as e:
+            logger.error(e.response.status_code)
+            logger.error(e.response.text)
+            if e.response.status_code == 400:
+                logger.warning(
+                    f"None of the paths {paths} were found in repository {repository['name']}"
+                )
+            return
+        except Exception as e:
+            logger.error(
+                f"Unexpected error processing files in {repository['name']}: {e}"
+            )
+            return
+
+    async def download_single_file(
+        self, file: dict[str, Any], repository: dict[str, Any], branch: str
+    ) -> dict[str, Any] | None:
+        if not file:
+            return None
+
+        if file.get("gitObjectType") != "blob":
+            return None
+
+        file_path = file["path"].lstrip("/")
+        content = await self.get_file_by_branch(file_path, repository["id"], branch)
+
+        if not content:
+            return None
+
+        file_size = len(content)
+        if file_size > MAX_ALLOWED_FILE_SIZE_IN_BYTES:
+            logger.warning(f"Skipping large file {file_path} ({file_size} bytes)")
+            return None
+
+        file_obj = {
+            "path": file_path,
+            "objectId": file["objectId"],
+            "size": file_size,
+            "isFolder": False,
+            "commitId": file.get("commitId"),
+            **file.get("contentMetadata", {}),
+        }
+
+        processed_file = await generate_file_object_from_repository_file(
+            file_obj, content, repository
+        )
```

Comment:
I think you either need to define a schema, or pass it as params or enrich some of it afterwards